PowerShellApp/DescriptiveDataUpload.ps1

```diff
@@ -0,0 +1,149 @@
+                $appToken = $TokenResult.AccessToken
+        }
+        catch {
+                Write-Error $_
+        }
+        
+        return $appToken;
+}
+
+function GetAppTokenFromClientCertificate ( [string] $ClientId, [string]$certificateName, [string] $TenantId ) {
+
+       
+        $appToken = ""
+        $certificate = FindCertificate $certificateName
+        try {
+               
+                $app = [Microsoft.Identity.Client.ConfidentialClientApplicationBuilder]::Create($ClientId).WithCertificate($certificate).WithAuthority($("$loginURL/$TenantId")).Build()
+                $TokenResult = $app.AcquireTokenForClient($Scopes).ExecuteAsync().Result;
+                $appToken = $TokenResult.AccessToken
+        }
+        catch {
+                Write-Error $_
+        }
+        
+        return $appToken;
+}
+
+
+
+$ClientId = Read-Host -Prompt "AppId/Client ID"
+ValidateString $ClientId
+$pathToZippedFile = Read-Host -Prompt "Please enter the absolute path to the zipped file you wish to upload"
+ValidateString $pathToZippedFile
+$TenantId = Read-Host -Prompt "Azure Active Directory (AAD) Tenant ID"
+ValidateString $TenantId
+$novaScaleUnit = Read-Host -Prompt "Scale unit associated with the AAD Tenant ID"
+ValidateString $novaScaleUnit
+
+if (-NOT(IsGuid $ClientId) -or -NOT(IsGuid $TenantId)) {
+        Write-Host   "The appId and/or the tenantId is not a valid Guid.`nPlease go through the process again to upload your file." -ForegroundColor Red
+        exit 0
+}       
+$option = Read-Host -Prompt "Please type 1 if you wish to provide certificate name and type 2 if you wish to provide client secret"
+$appToken
+if($option.Equals("1"))
+{
+        $certificateName = Read-Host -Prompt "Certificate name for your registered application"
+        ValidateString $certificateName
+        $appToken = GetAppTokenFromClientCertificate $ClientId $certificateName $TenantId 
+}
+else 
+{
+        $ClientSecret = Read-Host -Prompt "Client secret for your registered application"
+        ValidateString $ClientSecret
+        $appToken = GetAppTokenFromClientSecret $ClientId $ClientSecret $TenantId 
+}
+
+Write-Host $appToken
+$apiToAccess = $NovaPpeApi + $TenantId + "/ingress/connectors/HR/ingestions/fileIngestion"
+
+
+try {
+        $client = New-Object System.Net.Http.HttpClient
+        $client.DefaultRequestHeaders.Add('x-nova-scaleunit', $novaScaleUnit);
+        $client.DefaultRequestHeaders.Add("Authorization", "Bearer " + $appToken);
+        $content = New-Object System.Net.Http.MultipartFormDataContent
+        $fileStream = [System.IO.File]::OpenRead($pathToZippedFile)
+        $fileName = [System.IO.Path]::GetFileName($pathToZippedFile)
+        $fileContent = New-Object System.Net.Http.StreamContent($fileStream)
+        $content.Add($fileContent, "info", $fileName)
+        $result = $client.PostAsync($apiToAccess, $content).Result;
```

Comment:
Could you do one test in PROD using Powershell? https://microsoft.sharepoint-df.com/teams/WPAGoesBig/_layouts/15/Doc.aspx?sourcedoc={ec00bf8e-a063-46d4-8229-08681e286395}&action=edit&wd=target%28How%20to%20-%20PRD%20test%20APIs.one%7C6038555f-4e74-4336-ae21-499a3a7c84cb%2F%29&wdorigin=717

Reply:
^Ensure you are uploading file that will run into a Validation Failure - we don't want to overwrite anything in PROD. (Instructions mention this)